### DISCUSSION TITLE
Issue on connecting multiple UEs

### DISCUSSION THREAD

**Discussion by Ali78k** [srsran/srsRAN_Project#758] [General Help] [answered]
Hi community! I have 2 questions:
1- When I run one UE in ZMQ mode, a tun interface is created. But when I run the multiple UE Scenario, I don't see any tun interface.
how can I generate traffic for each UE?
![Screenshot from 2024-08-04 13-08-58](https://github.com/user-attachments/assets/f32b2efe-e9fa-440c-931f-6e15a41b1c91)
![Screenshot from 2024-08-04 13-10-42](https://github.com/user-attachments/assets/58dc99fe-704a-422d-b3f6-258675ba1026)

2- I've also expanded the scenario for 10 UEs. Although I imitated the gnuradio pattern and set proper configurations for UEs and open5Gs, UEs doesn't establish PDU sessions (except one of them).
![Screenshot from 2024-08-04 13-21-17](https://github.com/user-attachments/assets/a62870a4-a4ff-4f1b-b048-1ebeac973ad2)
![Screenshot from 2024-08-04 13-23-11](https://github.com/user-attachments/assets/7dbaff02-af8a-452f-9165-413d00ae60a3)

Any idea?

Logs were attached as pictures!

Thanks in advance!

**Comment by pgawlowicz** [comment]
Please follow this tutorial: https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#multi-ue-emulation

It looks like all UEs use the same preamble index, and cannot resolve the RACH contention.

**Reply by Ali78k** [reply]
Thank you @pgawlowicz . I've Already followed the instruction!
How can I check preamble index in UEs' config files? I couldn't find this attribute on the [UE sample conf](https://github.com/srsRAN/srsRAN_4G/blob/master/srsue/ue.conf.example)

**Reply by Ali78k** [reply]
Dear @pgawlowicz . I was able to run the simulated UE scenario before, but now it's not functioning correctly. The main issue is that no TUN interfaces are up when I start the srsUE!

**Reply by pgawlowicz** [reply]
You cannot set the preamble index in the UE config, but you can configure a number of available preambles in gnb config in the following way:
```
  prach:
    prach_config_index: 1           # Sets PRACH config to match what is expected by srsUE
+   total_nof_ra_preambles: 63      # Sets number of available PRACH preambles
```

The TUN interface is configured when a UE gets assigned an IP address. 
Please share your gnb and srsue logs, so we can check.

**Reply by Ali78k** [reply]
Many thanks @pgawlowicz .
Log files for gNB, UE , and CN are attached!
It is important to say that I've used the sample configuration files in [this tutorial](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#zeromq-based-setup).




[open5gs.log](https://github.com/user-attachments/files/16577851/open5gs.log)
[gnb.log](https://github.com/user-attachments/files/16577898/gnb.log)
[ue.log](https://github.com/user-attachments/files/16577900/ue.log)

![Screenshot from 2024-08-12 09-07-43](https://github.com/user-attachments/assets/edb053ec-f246-4aab-a9ba-b0d93f35b1ed)

Ali

**Reply by pgawlowicz** [reply]
if you want to ping an external IP from the srsUE, you need to add a default route:
```
sudo ip netns exec ue1 ip route add default via 10.45.1.1 dev tun_srsue
```
and enable IP forwarding on the PC running open5gs:
```
sudo sysctl -w net.ipv4.ip_forward=1
sudo iptables -t nat -A POSTROUTING -o <IFNAME> -j MASQUERADE
```
The above steps are described in the tutorial you follow.

**Reply by Ali78k** [reply]
Thanks for your quick reply @pgawlowicz .
I've already executed these commands, but the first one encountered an issue due to the TUN interface not being up. Please refer to the `ip addr show` command output in the image I attached in the previous comment.

**Reply by pgawlowicz** [reply]
could  you check
```
sudo ip netns exec ue1 route -n
```

**Reply by Ali78k** [reply]
here
```bash
root@ric:~# sudo ip netns exec ue1 route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
10.45.1.0       0.0.0.0         255.255.255.0   U     0      0        0 tun_srsue

```

**Reply by Ali78k** [reply]
Dear @pgawlowicz . I run `sudo ip ro add 10.45.0.0/16 via 10.53.1.2` as instructed in tutorial. This time, the command
```bash
sudo ip netns exec ue1 ip route add default via 10.45.1.1 dev tun_srsue
```
executed without issues, but the output was different from what was shown in the tutorial (it is like previous comment).

Edit: Now it is like the tutorial:

```bash
root@ric:~# sudo ip netns exec ue1 route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         10.45.1.1       0.0.0.0         UG    0      0        0 tun_srsue
10.45.1.0       0.0.0.0         255.255.255.0   U     0      0        0 tun_srsue
```
but there is not any tun_srsue yet
```bash
root@ric:~# ifconfig -a | grep tun
 
```

**Reply by pgawlowicz** [reply]
tun_srsue interface should be inside ue1:
```
sudo ip netns exec ue1 ifconfig
```

**Reply by Ali78k** [reply]
Many thanks @pgawlowicz . I can now ping `8.8.8.8` or any other servers by IP through `tun_srsue` interface, but I can't ping domain names. How can I fix it?

**Reply by pgawlowicz** [reply]
you need to configure DNS address in ue1 network namespace.

**Reply by Ali78k** [reply]
it is working now
Thank you so much @pgawlowicz  for your quick and helpful response—I really appreciate your guidance!
If you agree, I would like to keep this discussion in the 'open' state until I have also checked the multi UE scenario.

**Reply by Ali78k** [reply]
> You cannot set the preamble index in the UE config, but you can configure a number of available preambles in gnb config in the following way:
> 
> ```
>   prach:
>     prach_config_index: 1           # Sets PRACH config to match what is expected by srsUE
> +   total_nof_ra_preambles: 63      # Sets number of available PRACH preambles
> ```
> 
> The TUN interface is configured when a UE gets assigned an IP address. Please share your gnb and srsue logs, so we can check.

When I add `total_nof_ra_preambles: 63`, the gNB cannot run:
```bash
root@ric:~/newsrs/srsRAN_Project/build/apps/gnb# ./gnb -c multi_zmq.conf 

--== srsRAN gNB (commit 4ac5300d4) ==--

Total nof. RA preambles (63) should be a multiple of the number of SSBs per RACH occasion (1.0).
srsRAN ERROR: Invalid configuration detected.
```

I've cloned the main branch.

**Reply by pgawlowicz** [reply, marked as answer]
Thanks for pointing this issue out. The config has to be updated as the code was updated.
Please try with:
```
  prach:
    prach_config_index: 1
    total_nof_ra_preambles: 64
    nof_ssb_per_ro: 1
    nof_cb_preambles_per_ssb: 64
```

**Reply by Ali78k** [reply]
Thanks so much for your help @pgawlowicz  ! I’ve successfully set up and tested the project with ten UEs. Your guidance was really valuable.
